File: js/arrays_strings/container_with_most_water.js
const containerWithMostWater = (height) => {
    let left = 0;
    let right = height.length -1;
    let maxArea = 0;
    while(left < right){
        length = Math.min(height[left], height[right]);
        width = right - left;
        area = length * width;        
        maxArea = Math.max(maxArea, area);
        if(height[left] < height[right]){
            left++;
        } else{
            right--;
        }
    }
    return maxArea;
}
height = [2,3,4,5,18,17,6];
console.log(containerWithMostWater(height))

File: python/arrays_strings/container_with_most_water.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        maxArea = 0
        while(l < r):
            length = min(height[l], height[r])
            width = r-l
            area = width*length
            maxArea = max(maxArea, area)
            if(height[l] < height[r]):
                l += 1
            else:
                r -= 1
        return maxArea


s = Solution()
height = [1, 8, 6, 2, 5, 4, 8, 3, 7]
ans = s.maxArea(height)
print(ans)